README.md

```diff
@@ -1,407 +1,436 @@
-```
-
-Searches the vim oldfiles. See `:help v:oldfiles`
-
-```lua
-require'telescope.builtin'.quickfix{}
-```
-
-Search on the quickfix. See `:help quickfix`
-
-```lua
-require'telescope.builtin'.loclist{}
-```
-
-Search on the current window's location list.
-
-```lua
-require'telescope.builtin'.command_history{}
-```
-
-Search the vim command history.
-
-```lua
-require'telescope.builtin'.buffers{
-    -- Optional
-    -- show_all_buffers = true -- Show unloaded buffers aswell
+-- lua/telescope/pickers.lua
+Picker:new{
+  prompt_title       = "", -- REQUIRED
+  finder             = FUNCTION, -- see lua/telescope/finder.lua
+  sorter             = FUNCTION, -- see lua/telescope/sorter.lua
+  previewer          = FUNCTION, -- see lua/telescope/previewer.lua
+  selection_strategy = "reset", -- follow, reset, row
+  border             = {},
+  borderchars        = {"─", "│", "─", "│", "┌", "┐", "┘", "└"},
+  preview_cutoff     = 120,
 }
 ```
 
-Search on vim buffers list.
+##### Examples
 
-#### LSP
+###### Override mappings
 
-```lua
-require'telescope.builtin'.lsp_references{}
-```
-
-Search on LSP references.
-
-```lua
-require'telescope.builtin'.lsp_document_symbols{}
-```
-
-Search on LSP Document Symbols in the current document.
+To override only *some* of the default mappings, you can use the
+`attach_mappings` key in the `setup` table. For example:
 
 ```lua
-require'telescope.builtin'.lsp_workspace_symbols{}
-```
-
-Search on all workspace symbols.
-
-#### Treesitter
-
-```lua
-require'telescope.builtin'.treesitter{
-  -- Optional
-  -- bufnr = Buffer number
-}
-```
-
-Search on function names, variables, from Treesitter!
-
-Note: Requires nvim-treesitter
-#### Telescope
-
-```lua
-require'telescope.builtin'.planets{}
-```
-
-Use the telescope.
-
-## Themes
-
-Common groups of settings can be setup to allow for themes. We have some built in themes but are looking for more cool options. 
-
-### Dropdown
-
-![Dropdown Theme](https://i.imgur.com/SorAcXv.png)
-
-```vim
-nnoremap <Leader>f :lua require'telescope.builtin'.find_files(require('telescope.themes').get_dropdown({}))<cr>
-```
-
-Then you can put your configuration into `get_dropdown({})`
+function my_custom_picker(results)
+  pickers.new(opts, {
+    prompt_title = 'Custom Picker',
+    finder = finders.new_table(results),
+    sorter = sorters.fuzzy_with_index_bias(),
+    attach_mappings = function(_, map)
+      -- Map "<CR>" in insert mode to the funciton, actions.set_command_line
+      map('i', '<CR>', actions.set_command_line)
 
-```vim
-nnoremap <Leader>f :lua require'telescope.builtin'.find_files(require('telescope.themes').get_dropdown({ winblend = 10 }))<cr>
+      return true
+    end,
+  }):find()
+end
 ```
 
-Themes should work with every `telescope.builtin` function.  
+###### Planets example 
 
-If you wish to make theme, check out `lua/telescope/themes.lua`. If you need more features, make an issue :).
+see `lua/builtins.lua/planents`
```

Comment:
Why? Why in API? And i think the path should be wrong.

Reply:
Done